File: backend/models/project.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Project struct {
	Id             primitive.ObjectID `bson:"_id" json:"id,omitempty"`
	CommitCount    int                `bson:"commit_count" json:"commitCount,omitempty"`
	DeveloperCount int                `bson:"developer_count" json:"developerCount,omitempty"`
	FileCount      int                `bson:"file_count" json:"fileCount,omitempty"`
	LastScanDate   time.Time          `bson:"last_scan_date" json:"lastScanDate,omitempty"`
	Name           string             `bson:"name" json:"name,omitempty"`
	Url            string             `bson:"url" json:"url,omitempty"`
}

File: miner/models/project.py
from datetime import datetime
from typing import List, Optional, TypedDict


class Project(TypedDict):
    id: Optional[str]
    name: str
    commit_count: int
    developer_count: int
    file_count: int
    relevant_file_endings: List[str]
    last_scan_date: datetime
    url: str
    folder_depth: int
    issue_count: int
    pr_count: int
    jira_project_slug: str
    bitbucket_project_name: str
    bitbucket_repo_slug: str
